lightning/src/onion_message/messenger.rs

```diff
@@ -980,24 +973,25 @@ where
 					"Received an onion message with path_id {:02x?} and {} reply_path: {:?}",
 					path_id, if reply_path.is_some() { "a" } else { "no" }, message);
 
+				let message_type = message.msg_type();
 				match message {
 					ParsedOnionMessageContents::Offers(msg) => {
-						let response = self.offers_handler.handle_message(msg);
-						self.handle_onion_message_response(
-							response, reply_path, format_args!(
-								"when responding to Offers onion message with path_id {:02x?}",
-								path_id
-							)
-						);
+						let responder = reply_path.map(|path| Responder::new(path));
+						let response_instructions = self.offers_handler.handle_message(msg, responder);
+						self.handle_onion_message_response(response_instructions, format_args!(
+							"when responding to {} onion message with path_id {:02x?}",
+							message_type,
```

Comment:
If we aren't reusing this string literal, then the new `msg_type` method isn't useful.

Reply:
I am not sure I understand your comment correctly.
message_type is only used once in the handle_message_function currently (that is in `format_args!()`)
In the follow-up (#2996 )I have passed the `message_type` as an input parameter in `handle_onion_message_response`, where it can be reused if necessary.

So it would be great if you could help me understand how should I be reusing `message_type` in this PR. Thanks!

Reply:
Convert `msg_type()` to the associated function as per our offline discussion in [6109cd0](https://github.com/lightningdevkit/rust-lightning/commit/6109cd0c2e32c5e42f6d5bb8a30ebdd2dde8b350)
Thanks!